### DISCUSSION TITLE
What is the ideal data structure to send task/data between the async context and a dedicated thread?

### DISCUSSION THREAD

**Discussion by byte-sourcerer** [tokio-rs/tokio#6522] [Q&A] [answered]
I read the blog [Async: What is blocking?](https://ryhl.io/blog/async-what-is-blocking/), which advocates the concept of employing a long running dedicated thread to handle blocking task. Consequently, this prompts the question - what would be the ideal data structure for passing tasks or data to the long running dedicated thread?

I have the following options:

1. Utilizing `tokio::sync::mpsc::channel` along with `blocking_send`/`blocking_rcv` in the dedicated thread.
2. Using crates like [crossfire](https://github.com/qingstor/crossfire-rs).

**Comment by Darksonn** [comment, marked as answer]
I would generally recommend using things from `tokio::sync` unless you have a good reason not to. So in this case, I recommend `tokio::sync::mpsc` over some other mpsc channel.
